Source/WebCore/editing/markup.cpp

```diff
@@ -358,6 +358,27 @@ class StyledMarkupAccumulator final : public MarkupAccumulator {
         m_reversedPrecedingMarkup.append("<head>"_s);
     }
 
+    void wrapInBodyTagWithDirectionAttributeIfNecessary(const VisiblePosition& start, const VisiblePosition& end)
+    {
+        if (!inSameParagraph(start, end))
+            return;
+
+        if (m_highestNodeToBeSerialized && m_highestNodeToBeSerialized->hasTagName(bodyTag))
+            return;
+
+        auto block = enclosingBlock(start.deepEquivalent().protectedContainerNode());
+        if (!block || block != enclosingBlock(end.deepEquivalent().protectedContainerNode()))
+            return;
+
+        auto renderer = block->renderer();
+        if (!renderer)
+            return;
+
+        auto directionValue = renderer->writingMode().bidiDirection() == TextDirection::LTR ? "ltr"_s : "rtl"_s;
+        m_reversedPrecedingMarkup.append(makeString("<body dir=\""_s, WTFMove(directionValue), "\">"_s));
+        append("</body>"_s);
```

Comment:
Nit either way, but why not just put `</body>` in the previous line?

Reply:
oh, my bad, I didn't pay attention to the markup.

Reply:
No worries! Will keep this as-is. Thanks for the review!